src/Stratis.Features.SQLiteWalletRepository/Tables/HDTransactionData.cs

```diff
@@ -222,5 +223,61 @@ FROM    HDTransactionData
+        internal static IEnumerable<FlattenedHistoryItem> GetHistory(DBConnection conn, int walletId, int accountIndex, int limit, int offset, string txId)
+        {
+            string strLimit = DBParameter.Create(limit);
+            string strOffset = DBParameter.Create(offset);
+            string strWalletId = DBParameter.Create(walletId);
+            string strAccountIndex = DBParameter.Create(accountIndex);
+            string strTransactionId = DBParameter.Create(txId);
+
+            var result = conn.Query<FlattenedHistoryItem>($@"
+                    SELECT 
+                        t.OutputTxId as Id, 
+                        CASE 
+                            WHEN t.OutputTxIsCoinbase = 0 AND t.AddressType = 0 THEN 0
+                            WHEN t.OutputTxIsCoinbase = 0 AND t.AddressType = 1 THEN 1
+                            WHEN t.OutputTxIsCoinbase = 1 AND t.OutputIndex != 0 THEN 2
+                            WHEN t.OutputTxIsCoinbase = 1 AND t.OutputIndex = 0 THEN 3
+                        END Type,                    
+                        t.OutputTxTime as TimeStamp,
+                        CASE
+                            WHEN t.OutputTxIsCoinbase = 0 AND t.AddressType = 0 THEN t.Value
+                            WHEN t.OutputTxIsCoinbase = 0 AND t.AddressType = 1 THEN ((SELECT sum(tt.Value) FROM HDTransactionData tt WHERE tt.SpendTxId = t.OutputTxId) - t.Value)
+                            WHEN t.OutputTxIsCoinbase = 1 AND t.OutputIndex = 0 THEN t.Value
+                            WHEN t.OutputTxIsCoinbase = 1 AND t.OutputIndex != 0 THEN (SUM(t.Value) -
+                            (
+                                SELECT tt.Value
+                                FROM HDPayment p
+                                INNER JOIN HDTransactionData tt ON tt.OutputTxId = p.OutputTxId AND tt.OutputIndex = p.OutputIndex
+                                WHERE p.SpendTxId = t.OutputTxId AND p.SpendIsChange = 0
+                                LIMIT 1
+                            ))
+                        END Amount,
+                        sends.SpendValue as SendValue,
+                        sends.SpendScriptPubkey as SendToScriptPubkey,
+                        t.OutputBlockHeight as BlockHeight
+                    FROM 
+                        HDTransactionData AS t
+                    LEFT OUTER JOIN HDPayment sends on sends.SpendTxId = t.OutputTxId AND sends.SpendIsChange = 0 AND t.OutputTxIsCoinbase = 0 AND t.AddressType = 1
+                    LEFT OUTER JOIN HDAddress a on a.Pubkey = sends.SpendScriptPubkey
+                    WHERE 
+                        t.WalletId = {strWalletId} AND t.AccountIndex = {strAccountIndex} {((txId == null) ? "" : $@" AND t.OutputTxId = {strTransactionId}")}
+                    GROUP BY 
+                        t.OutputTxId
+                    ORDER BY
+                        t.OutputTxTime DESC
```

Comment:
Only receiving of funds is being ordered. Should the data be merged with a query getting all the spends ordered separately?

Reply:
@quantumagi this is very strange as the SQL i get back is perfectly ordered... I.e receives/sends/stakes/mines all in perfect date order? 🤔

Reply:
It seems we will only be displaying the receive date - so that alone would be an issue.